core/trino-main/src/main/java/io/trino/operator/scalar/ArrayVectorFunctions.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.function.Description;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlType;
+
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.StandardTypes.DOUBLE;
+import static io.trino.util.Failures.checkCondition;
+
+public final class ArrayVectorFunctions
+{
+    private ArrayVectorFunctions() {}
+
+    @Description("Calculates the euclidean distance between two vectors. This is same as L2 distance <-> in pgvector.")
+    @ScalarFunction
+    @SqlType(DOUBLE)
+    public static double euclideanDistance(@SqlType("array(real)") Block first, @SqlType("array(real)") Block second)
+    {
+        checkCondition(first.getPositionCount() == second.getPositionCount(), INVALID_FUNCTION_ARGUMENT, "The arguments must have the same length");
+
+        double sum = 0.0;
+        for (int i = 0; i < first.getPositionCount(); i++) {
+            double diff = REAL.getFloat(first, i) - REAL.getFloat(second, i);
```

Comment:
Do we need to cast one of the arguments to a double to get proper overflow handling ?
E.g. first value is Float.MAX_VALUE and second value is Float.MIN_VALUE